packages/core/lib/commands/test/run.js

```diff
@@ -68,7 +68,42 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url exists, then it connects to that external network.
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   * If development network exists, then it connects to the development network specified in the config.
+   */
+  if (
+    config.networks[config.network] &&
+    config.network == "test" &&
+    !config.networks[config.network]["provider"] &&
+    !config.networks[config.network]["host"] &&
+    !config.networks[config.network]["url"]
+  ) {
+    const ipcOptions = { network: "test" };
+    const port = await require("get-port")();
```

Comment:
`port` is used in the other branches and could be factored out so it is only initialized once.